api/pkg/apis/v1alpha1/providers/target/script/script.go

```diff
@@ -394,6 +394,14 @@ func (i *ScriptProvider) Apply(ctx context.Context, deployment model.DeploymentS
 			ret[k] = v
 		}
 	}
+
+	for _, v := range ret {
+		switch v.Status {
+		case v1alpha2.DeleteFailed, v1alpha2.ValidateFailed, v1alpha2.UpdateFailed:
+			err := v1alpha2.NewCOAError(errors.New(v.Message), "failed to execute script", v.Status)
```

Comment:
I think it is better to differentiate the error between execution with non-zero return and error reporting in output file. Is that better to change the error message to "executing script returned error output"?